src/main/java/com/box/sdk/BoxTransactionalAPIConnection.java

```diff
@@ -0,0 +1,166 @@
+package com.box.sdk;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+
+import com.eclipsesource.json.JsonObject;
+
+/**
+ * Represents an authenticated transactional connection to the Box API.
+ *
+ * <p>This class handles everything for transactional API that isn't already handled by BoxAPIConnection.</p>
+ */
+public class BoxTransactionalAPIConnection extends BoxAPIConnection {
+
+    private static final String SUBJECT_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token";
+    private static final String GRANT_TYPE = "urn:ietf:params:oauth:grant-type:token-exchange";
+
+    /**
+     * Constructs a new BoxTransactionalAPIConnection that authenticates with an access token.
+     * @param  accessToken a transactional auth access token.
+     */
+    public BoxTransactionalAPIConnection(String accessToken) {
+        super(accessToken);
+        super.setAutoRefresh(false);
+    }
+
+    /**
+     * Disabling an invalid constructor for Box Developer Edition.
+     * @param  clientID     the client ID to use when refreshing the access token.
+     * @param  clientSecret the client secret to use when refreshing the access token.
+     * @param  accessToken  an initial access token to use for authenticating with the API.
+     * @param  refreshToken an initial refresh token to use when refreshing the access token.
+     * @throws UnsupportedOperationException Box Transactional API does not support creating
+     *         a connection with these params
+     */
+    public BoxTransactionalAPIConnection(String clientID, String clientSecret, String accessToken,
+                                            String refreshToken) {
+        super(null);
+        throw new UnsupportedOperationException("This constructor is not available for BoxTransactionalAPIConnection.");
+    }
+
+    /**
+     * Disabling an invalid constructor for Box Developer Edition.
+     * @param  clientID     the client ID to use when exchanging the auth code for an access token.
+     * @param  clientSecret the client secret to use when exchanging the auth code for an access token.
+     * @param  authCode     an auth code obtained from the first half of the OAuth process.
+     * @throws UnsupportedOperationException Box Transactional API does not support creating
+     *         a connection with these params
+     */
+    public BoxTransactionalAPIConnection(String clientID, String clientSecret, String authCode) {
+        super(null);
```

Comment:
In Java (most languages?) constructors are never inherited or overriden. Which is why you have to call super() manually to chain constructors. So if you don't want BoxTransactionalAPIConnection being new-ed up with certain arguments, just don't add a constructor.